src/apps/notifications/components/NotificationDetails.tsx

```diff
@@ -0,0 +1,242 @@
+  TablePagination
+} from "@material-ui/core";
+import {
+  NotificationItemRow,
+  NotificationItem
+} from "../types";
+import { EnhancedNotificationSummaryTableHead } from "./EnhancedNotificationSummaryTableHead"
+
+interface Props {
+  open: boolean;
+  handleClose: () => void;
+  notification: NotificationItem;
+}
+
+const useStyles = makeStyles({
+  dialogTitle: {
+    textAlign: "center"
+  },
+  tableContainer: {
+    display: "contents"
+  },
+  tablePagination: {
+    bottom: "0",
+    position: "sticky",
+    backgroundColor: "#fafafa"
+  },
+  noPadding: {
+    padding: "0px"
+  }
+});
+
+const NotificationDetails: React.FC<Props> = ({ open, handleClose, notification }) => {
+
+  const classes = useStyles();
+
+  const getParentConceptIds = () => {
+    const meta = notification.meta;
+    if(meta && meta.length > 1) {
+      const requestPayload = meta[1];
+      const conceptIds = requestPayload.map((concept: { id: string }) => concept.id);
+      return conceptIds;
+    } else {
+      return [];
+    }
+  };
+  const parentConceptIds = getParentConceptIds();
+
+  const getConceptId = (row: NotificationItemRow) => {
+    return row.expression.split("/")[6];
+  };
+
+  const getConceptType = (row: NotificationItemRow) => {
+    const getActualIndex = (ele: string) => ele ===  getConceptId(row);
+    const actualIndex = parentConceptIds.findIndex(getActualIndex);
+    if (actualIndex > -1) {
+      return "Parent";
+    } else {
+      return "Dependent";
+    }
+  };
+
+  const getStatus = (row: NotificationItemRow) => {
+    return row.added ? "Imported" : "Skipped";
+  };
+
+  const getReason = (row: NotificationItemRow) => {
+    if (typeof row.message === "string") {
+      return row.message;
+    }
+    return row.message.join(" ");
+  };
+
+  const getDialogTitle = () => {
+    const dictionaryName = notification.meta ? notification.meta[0].split("/")[4] : "";
+
+    const sourceName = notification.result.length > 0 ? notification.result[0].expression.split("/")[4] : "";
+    return (
+        <Typography variant='h5'>
+          {dictionaryName} - Adding concepts from {sourceName}
+        </Typography>
+    );
+  };
+
+  const createData = (
+      conceptId: string,
+      conceptType: string,
+      status: string,
+      reasons: string,
+  ) => {
+    return { conceptId, conceptType, status, reasons };
+  };
+
+  const getSummaryRowsToDisplay = () => {
+    const conceptRows = notification.result.filter(row =>
+        row.expression.includes("/concepts/")
+    );
+    return conceptRows.map((row) => {
+      return createData(getConceptId(row), getConceptType(row), getStatus(row), getReason(row));
+    });
+  };
+
+  const summaryRowsToDisplay = getSummaryRowsToDisplay();
+
+  type Order = 'asc' | 'desc';
+
+  const [order, setOrder] = React.useState<Order>("desc");
+  const [orderBy, setOrderBy] = React.useState("status");
+  const handleRequestSort = (event: React.MouseEvent<unknown>, property: string) => {
```

Comment:
@tarunkumar-tw @rakesh-1010 can we move this sorting logic to a util file ?

Reply:
@binduak Moved sorting functions to utils file